.github/workflows/main.yml

```diff
@@ -53,6 +53,47 @@ jobs:
           echo $GITHUB_WORKSPACE
           docker run --privileged -e CI=true -t -v $GITHUB_WORKSPACE:/github/workspace docker.pkg.github.com/fwupd/fwupd/fwupd-${{matrix.os}}:latest
 
+  build-freebsd:
+    runs-on: macos-latest
+    name: build (FreeBSD-12.2)
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+    - name: Find tag
+      id: tagger
+      uses: jimschubert/query-tag-action@v1
+    - name: Build
+      id: test
+      uses: vmactions/freebsd-vm@v0.1.3
+      with:
+        usesh: true
+        mem: 8192
+        prepare: |
+          pkg install -y git python3 glib meson pkgconf gobject-introspection \
+            vala gtk-doc json-glib gpgme gnutls sqlite3 curl gcab libarchive \
+            help2man libelf libgpg-error gettext-tools gtk-update-icon-cache \
+            atk pango
+        sync: rsync
+        run: |
+          cd /usr
+          git clone https://github.com/3mdeb/freebsd-ports.git --depth 1 -b fwupd ports
```

Comment:
Just some comments on this:
1. how come gusb can't subproject?  If you could rely on subproject, then you can upstream those patches and depend on a newer tag or release potentially.
2. It would be nice if at least for CI that `sysutils/fwupd/*` was in upstream `contrib/`.  When we drop/add new options then that thing can be directly patched as part of CI.  Otherwise if we remove an option or change the default, CI might break for FreeBSD since we can't directly change it.  That's why we keep debian packaging and fedora packaging upstream for example.
3. Is that libjcat patch and libxmlb really needed?  They both look odd to me.

Reply:
@superm1 1. More information on this topic https://github.com/hughsie/libgusb/pull/10
2. Totally agree. After clone the freebsd-ports fwupd files could be replaced with the one from the `contrib/`.
3. Yes, otherwise meson configuration fails:
libxmlb:
```
Configuring config.h using configuration
Program python found: NO

meson.build:152:0: ERROR: python3 not found
```
libjcat
```
Program python (xml.etree.ElementTree, pkg_resources) found: NO

libjcat/meson.build:142:2: ERROR: python3 not found
```

Reply:
> meson.build:152:0: ERROR: python3 not found

Most unexpected. We should fix that upstream certainly.

Reply:
Okay, you can use subproject versions of master for everything now it it just works without patches.

Reply:
Until releases are tagged I guess that means moving subprojects/*.wrap over to master for those needed.

Reply:
hang on for now; i was going to do new releases of all the deps next week to keep this sane